tests/components/homeassistant/test_scene.py

```diff
@@ -258,3 +258,33 @@
         ("scene.scene_3", ["light.kitchen", "light.living_room"]),
     ):
         assert ha_scene.entities_in_scene(hass, scene_id) == entities
+
+
+async def test_config(hass):
+    """Test passing config in YAML."""
+    assert await async_setup_component(
+        hass,
+        "scene",
+        {
+            "scene": [
+                {
+                    "id": "scene_id",
+                    "name": "Scene Icon",
+                    "icon": "mdi:party",
+                    "entities": {"light.kitchen": "on"},
+                },
+                {
+                    "name": "Scene No Icon",
+                    "entities": {"light.kitchen": {"state": "on"}},
+                },
+            ]
+        },
+    )
+
+    icon = hass.states.get("scene.scene_icon")
+    assert icon is not None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5020228)